src/Symfony/Component/Console/Output/ConsoleOutput.php

```diff
@@ -51,6 +53,18 @@ public function __construct($verbosity = self::VERBOSITY_NORMAL, $decorated = nu
         if (null === $decorated) {
             $this->setDecorated($actualDecorated && $this->stderr->isDecorated());
         }
+
+        $this->outputReference = new OutputSectionReference();
+    }
+
+    /**
+     * Creates a new output section.
+     *
+     * @return ConsoleSectionOutput
+     */
+    public function section()
+    {
+        return new ConsoleSectionOutput($this->getStream(), $this->outputReference, $this->getVerbosity(), $this->isDecorated(), $this->getFormatter());
```

Comment:
This method should throw if `!$this->isDecorated()`

Reply:
@chalasr I don't think this should throw as it might cause issues in E.G a CI environment where the output should just continue printing without decoration. If this throws an exception, the commands would fail and not continue to run, which might not be the expected behaviour. I have instead added checks for decoration to skip the ANSI codes